src/renderer/redux/reducers/content.js

```diff
@@ -22,6 +22,21 @@ reducers[ACTIONS.FETCH_FEATURED_CONTENT_COMPLETED] = (state, action) => {
   });
 };
 
+reducers[ACTIONS.FETCH_FEATURED_CHANNELS_STARTED] = state =>
+  Object.assign({}, state, {
+    fetchingFeaturedChannels: true,
+  });
+
+reducers[ACTIONS.FETCH_FEATURED_CHANNELS_COMPLETED] = (state, action) => {
+  const { channels, success } = action.data;
+
+  return Object.assign({}, state, {
+    fetchingFeaturedChannels: false,
+    fetchingFeaturedChannelsFailed: !success,
+    featuredChannels: channels,
```

Comment:
This looks like a new copy of claims-in-channel data, which would be a DRY break. Use the same place that other channel claim data is stored. Please look at `ACTIONS.FETCH_CHANNEL_CLAIMS_COMPLETED`.

Reply:
That's what I was going for all week but couldn't get all the parts together. I could never get the component to re-render when the channels came in, or couldn't load from channels correctly while not interfering with the subscriptions page or having to reset before viewing either page to not mix featured and subscribed channels.

I pushed this because it worked. I will try more of that tomorrow and push.

Reply:
@kauffj a more thorough explanation:

`ACTIONS.FETCH_CHANNEL_CLAIMS_COMPLETED` stores claim data in `state.claims.claimsByChannel`

rewriting the reducer to store the data in `state.content.featuredUris` when requested by the Discover page rather than the Subscriptions page seems like it will add as much code and complexity as my approach of adding 3 `state.content` properties, 1 action, and 2 reducers

if the Discover page is putting featured channels in `state.claims.claimsByChannel` then both featured and subscribed channels will be displayed on the Subscriptions page

and on the Discover page if/when the users goes back

each page could filter out the other's claim data in the view but that seems hackish

not to mention this would also require adding all the subscription fetching code to the discover component, which is a huge mess with the following comment:

`// setHasFetchedSubscriptions is a terrible hack
// it allows the subscriptions to load correctly when refresing on the subscriptions page
// currently the page is rendered before the state is rehyrdated
// that causes this component to be rendered with zero savedSubscriptions
// we need to wait until persist/REHYDRATE has fired before rendering the page`

https://github.com/lbryio/lbry-app/blob/master/src/renderer/page/subscriptions/view.jsx

this is the road I went down--and ran into rendering issues I still haven't figured out--before becoming convinced that treating Featured Channels as its own feature worthy of its own place in the data model was simpler than trying to implement it using functions built for Featured Uris and Subscriptions

I may not need `fetchingFeaturedChannels` and am not yet using `fetchingFeaturedChannelsFailed` so this doesn't need to be a full copy of `featuredUris`'s data structure, I was just trying to get something that made sense on here for y'all to look at

I am happy to refactor the Subscriptions page but that seems like a bigger issue than Featured Channels

or I could still be missing a simple solution, this part of the app is still new to me

Reply:
@daovist 

1) Not storing the same data in two different places or two different ways is about as fundamental as it gets when it comes to DRY. It is exceedingly unlikely we're accept a solution that involves storing the same data in two places or in two different ways (caching for performance reasons is the only exception I can think of).

2) If a simpler data model comes at the cost at more code that is entirely acceptable. 

3) I suspect there is a misunderstanding (or someone else is doing something very wrong), if data additions to `claimsByChannel` are affecting subscriptions. `claimsByChannel` is where all data is stored related to claims in a channel. Accessing any channel page stores data there.

4) I'm not following a lot of your comments. I don't understand how this is related to subscriptions at all. I don't understand what you mean with your comments related to filtering. I strongly suspect this functionality should be doable with 0 additional actions and 0 additional Redux state values.